iroh-net-report/src/reportgen.rs

```diff
@@ -898,8 +896,15 @@ async fn run_quic_probe(
         ProbeProto::QuicIpv6 => debug_assert!(relay_addr.is_ipv6()),
         _ => debug_assert!(false, "wrong probe"),
     }
-    // TODO(ramfox): what to put here if no host is given?
-    let host = url.host_str().unwrap_or("localhost");
+    let host = match url.host_str() {
+        Some(host) => host,
+        None => {
+            return Err(ProbeError::Error(
+                anyhow!("URL must have 'host' to use QUIC address discovery probes"),
+                probe.clone(),
+            ));
+        }
+    };
```

Comment:
FYI this is a prime place for using the fancy new-ish syntax of `let Some(host) = url.host_str() else { return ... };`  Not important enough to follow up on though.